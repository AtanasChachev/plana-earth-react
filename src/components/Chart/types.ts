import { ProductStatisticsValue } from 'models/products';

export interface ChartProps {
  data: ProductStatisticsValue[];
}