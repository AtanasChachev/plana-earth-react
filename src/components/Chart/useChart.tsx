import { State } from 'models/store';
import { useSelector } from 'react-redux';
import { ChartStateData } from 'models/chart';

type Return = {
  data: ChartStateData;
};

export const useChart = (): Return => {
  const { data } = useSelector((state: State) => state.chartState);

  return {
    data,
  };
};