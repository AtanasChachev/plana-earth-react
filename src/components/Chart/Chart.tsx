import { useChart } from './useChart';
import { ResponsiveContainer, LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip } from 'recharts';
import { SETTINGS } from 'config/settings';
import './Chart.scss';

const Chart = (): JSX.Element => {
  const { data } = useChart();

  return (
    <div className="holder">
      <ResponsiveContainer width="100%" height={400}>
        <LineChart data={data}>
          <Line type="monotone" dataKey="average" stroke={SETTINGS.chartColors.average}/>
          <Line type="monotone" dataKey="min" stroke={SETTINGS.chartColors.min} />
          <Line type="monotone" dataKey="standard deviation" stroke={SETTINGS.chartColors.standarddeviation} />

          <CartesianGrid stroke={SETTINGS.theme.palette.primary.main} strokeDasharray="5 5" />
          <XAxis dataKey="time" />
          <YAxis />
          <Tooltip />
        </LineChart>
      </ResponsiveContainer>
    </div>
  );
};

export { Chart };