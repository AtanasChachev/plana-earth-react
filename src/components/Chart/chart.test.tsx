import Chart from './Chart';
import { render } from '@testing-library/react';

const mockData = [
  {
    'time': '2022-10-04',
    'average': 1923.1207275390625,
    'count': 1,
    'max': 1923.1207275390625,
    'min': 1923.1207275390625,
    'standarddeviation': 15.24512392132,
  },
  {
    'time': '2022-10-26',
    'average': 1860.1245422363281,
    'count': 4,
    'max': 1873.88232421875,
    'min': 1845.958251953125,
    'standarddeviation': 12.96518628068393,
  },
  {
    'time': '2022-11-23',
    'average': 1897.4003228081597,
    'count': 27,
    'max': 1920.560302734375,
    'min': 1872.42626953125,
    'standarddeviation': 11.01504923711368,
  },
];

const setup = () => {
  const { asFragment } = render(
      <Chart data={mockData}/>,
  );

  return {
    asFragment,
  };
};

describe('<Chart />', () => {
  test('<Chart /> should render properly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});