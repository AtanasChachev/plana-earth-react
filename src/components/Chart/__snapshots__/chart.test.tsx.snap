// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart /> <Chart /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="holder"
  >
    <div
      class="recharts-responsive-container"
      style="width: 100%; height: 400px;"
    />
  </div>
</DocumentFragment>
`;
